### PR TITLE
feat: support the 'Description' resource attribute

### PR DESCRIPTION
**Pull request by skinny85**
One more resource attribute that we missed,
and that is needed for cfn-include to be able to handle ingesting all templates.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts [hunk 1]
```diff
@@ -589,7 +589,7 @@ export class CfnInclude extends core.CfnElement {
 
     // fail early for resource attributes we don't support yet
     const knownAttributes = [
-      'Type', 'Properties', 'Condition', 'DependsOn', 'Metadata', 'Version',
+      'Type', 'Properties', 'Condition', 'DependsOn', 'Metadata', 'Version', 'Description',
       'CreationPolicy', 'UpdatePolicy', 'DeletionPolicy', 'UpdateReplacePolicy',
```
Comment (shivlaks) [inline packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts:593]: might be good to make this list alphabetical for easier reading.
don't mind if the set of policy properties follow the regular attributes.

File: packages/@aws-cdk/core/lib/cfn-resource.ts [hunk 2]
```diff
@@ -438,6 +439,16 @@ export interface ICfnResourceOptions {
    */
   version?: string;
 
+  /**
+   * The description of this resource.
+   * Used for informational purposes only,
+   * is not processed in any way
+   * (and stays with the CloudFormation template,
+   * is not passed to the udnerlying resource,
```
Comment (shivlaks) [inline packages/@aws-cdk/core/lib/cfn-resource.ts:447]:
```suggestion
   * is not passed to the underlying resource,
```

Comment (shivlaks) [inline packages/@aws-cdk/core/lib/cfn-resource.ts:447]: what's with the line length? is it intentionally shorter? i feel like some of it flows better if it's on the same line :)

**Review by shivlaks — APPROVED** [review]
looks good! some small suggestions - feel free to make a call on them and drop the `do-not-merge` as you see fit!

File: packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts:593]: Sure!

```suggestion
      'Condition', 'DependsOn', 'Description', 'Metadata', 'Properties', 'Type', 'Version',
      'CreationPolicy', 'DeletionPolicy', 'UpdatePolicy', 'UpdateReplacePolicy'
```

File: packages/@aws-cdk/core/lib/cfn-resource.ts (hunk 2 above)
Comment (skinny85) [reply packages/@aws-cdk/core/lib/cfn-resource.ts:447]: Done!

```suggestion
  /**
   * The description of this resource.
   * Used for informational purposes only, is not processed in any way
   * (and stays with the CloudFormation template, is not passed to the underlying resource,
```

**Pull request merged**